common/interfaces/card_interface/nfc.c

```diff
@@ -177,8 +178,10 @@ ret_code_t nfc_wait_for_card(const uint16_t wait_time) {
 }
 
 void nfc_card_presence_detect() {
-  if (nfc_wait_for_card(DEFAULT_NFC_TG_INIT_TIME) == STM_SUCCESS)
+  if (nfc_wait_for_card(DEFAULT_NFC_TG_INIT_TIME) == STM_SUCCESS) {
+    nfc_set_card_detect_event();
     nfc_tapped = true;
```

Comment:
will this be a TODO? To remove after refactor.

Reply:
Removed in https://github.com/Cypherock/x1_wallet_firmware/pull/216
Can add a todo if decide to not merge it.